starknet_py/utils/crypto/transaction_hash.py

```diff
@@ -0,0 +1,96 @@
+from enum import Enum
+from typing import Sequence
+
+from starkware.starknet.core.os.class_hash import compute_class_hash
+
+from starknet_py.common import int_from_bytes
+from starknet_py.constants import QUERY_VERSION_BASE
+from starknet_py.net.client_models import ContractClass
+from starknet_py.utils.crypto.facade import compute_hash_on_elements
+
+
+class TransactionHashPrefix(Enum):
+    DECLARE = int_from_bytes(b"declare")
+    DEPLOY = int_from_bytes(b"deploy")
+    DEPLOY_ACCOUNT = int_from_bytes(b"deploy_account")
+    INVOKE = int_from_bytes(b"invoke")
+    L1_HANDLER = int_from_bytes(b"l1_handler")
+
+
+# pylint: disable=too-many-arguments
+def compute_transaction_hash(
+    tx_hash_prefix: TransactionHashPrefix,
+    version: int,
+    contract_address: int,
+    entry_point_selector: int,
+    calldata: Sequence[int],
+    max_fee: int,
+    chain_id: int,
+    additional_data: Sequence[int],
+) -> int:
+    calldata_hash = compute_hash_on_elements(data=calldata)
+    data_to_hash = [
+        tx_hash_prefix.value,
+        version,
+        contract_address,
+        entry_point_selector,
+        calldata_hash,
+        max_fee,
+        chain_id,
+        *additional_data,
+    ]
+
+    return compute_hash_on_elements(
+        data=data_to_hash,
+    )
+
+
+def compute_deploy_account_transaction_hash(
+    version: int,
+    contract_address: int,
```

Comment:
Wouldn't it make more sense not to take `contract_address` as a parameter and calculate it within the method instead?

Reply:
From cairo-lang code it seems that `DeployAccount.calculate_hash` method has `0` hardcoded for `deployer_address`. I'm not sure if it makes sense to expose this parameter specifically in the `DEPLOY_ACCOUNT` tx context.

Reply:
On the second though, perhaps it makes sense to just do it the way cairo-lang does it. This can be treated as lower level function so maybe it makes sense to not have this kind of implicit logic.